RecoLocalCalo/HcalRecAlgos/plugins/HcalRecAlgoESProducer.cc

```diff
@@ -94,5 +96,19 @@ HcalRecAlgoESProducer::produce(const HcalSeverityLevelComputerRcd& iRecord)
    return myComputer ;
 }
 
+void HcalRecAlgoESProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+    desc.add<uint32_t>("phase",0);
+    desc.add<std::vector<std::string>>("RecoveredRecHitBits");
+    desc.add<std::vector<std::string>>("DropChannelStatusBits");
+    descriptions.add("hcalRecAlgos",desc);
+
+    edm::ParameterSetDescription desc_sevlvl;
+    desc_sevlvl.add<uint32_t>("Level");
+    desc_sevlvl.add<std::vector<std::string>>("RecHitFlags");
+    desc_sevlvl.add<std::vector<std::string>>("ChannelStatus");
+    desc_sevlvl.addVPSet("SeverityLevels",desc_sevlvl);
+}
```

Comment:
You can use ```edmPythonConfigToCppValidation``` to generate the C++ fillDescriptions from an existing cfi file.
For example, using what is currently in the release (and editing the cfi file to be an EDProducer rather than an ESProducer as apparently ```edmPythonConfigToCppValidation``` only looks for EDProducers in the cfi file),
the result is:
#include <FWCore/ParameterSet/interface/ConfigurationDescriptions.h>
#include <FWCore/ParameterSet/interface/ParameterSetDescription.h>

void
HcalRecAlgoESProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
  // hcalRecAlgos
  edm::ParameterSetDescription desc;
  desc.add<std::vector<std::string>>("RecoveredRecHitBits", {
    "TimingAddedBit",
    "TimingSubtractedBit",
  });
  {
    edm::ParameterSetDescription vpsd1;
    vpsd1.add<std::vector<std::string>>("RecHitFlags", {
      "",
    });
    vpsd1.add<std::vector<std::string>>("ChannelStatus", {
      "",
    });
    vpsd1.add<int>("Level", 0);
    std::vector<edm::ParameterSet> temp1;
    temp1.reserve(8);
    {
      edm::ParameterSet temp2;
      temp2.addParameter<std::vector<std::string>>("RecHitFlags", {
        "",
      });
      temp2.addParameter<std::vector<std::string>>("ChannelStatus", {
        "",
      });
      temp2.addParameter<int>("Level", 0);
      temp1.push_back(temp2);
    }
    {
      edm::ParameterSet temp2;
      temp2.addParameter<std::vector<std::string>>("RecHitFlags", {
        "",
      });
      temp2.addParameter<std::vector<std::string>>("ChannelStatus", {
        "HcalCellCaloTowerProb",
      });
      temp2.addParameter<int>("Level", 1);
      temp1.push_back(temp2);
    }
    {
      edm::ParameterSet temp2;
      temp2.addParameter<std::vector<std::string>>("RecHitFlags", {
        "HSCP_R1R2",
        "HSCP_FracLeader",
        "HSCP_OuterEnergy",
        "HSCP_ExpFit",
        "ADCSaturationBit",
        "HBHEIsolatedNoise",
        "AddedSimHcalNoise",
      });
      temp2.addParameter<std::vector<std::string>>("ChannelStatus", {
        "HcalCellExcludeFromHBHENoiseSummary",
      });
      temp2.addParameter<int>("Level", 5);
      temp1.push_back(temp2);
    }
    {
      edm::ParameterSet temp2;
      temp2.addParameter<std::vector<std::string>>("RecHitFlags", {
        "HBHEHpdHitMultiplicity",
        "HBHEPulseShape",
        "HOBit",
        "HFDigiTime",
        "HFInTimeWindow",
        "ZDCBit",
        "CalibrationBit",
        "TimingErrorBit",
        "HBHEFlatNoise",
        "HBHESpikeNoise",
        "HBHETriangleNoise",
        "HBHETS4TS5Noise",
        "HBHENegativeNoise",
        "HBHEOOTPU",
      });
      temp2.addParameter<std::vector<std::string>>("ChannelStatus", {
        "",
      });
      temp2.addParameter<int>("Level", 8);
      temp1.push_back(temp2);
    }
    {
      edm::ParameterSet temp2;
      temp2.addParameter<std::vector<std::string>>("RecHitFlags", {
        "HFLongShort",
        "HFPET",
        "HFS8S1Ratio",
      });
      temp2.addParameter<std::vector<std::string>>("ChannelStatus", {
        "",
      });
      temp2.addParameter<int>("Level", 11);
      temp1.push_back(temp2);
    }
    {
      edm::ParameterSet temp2;
      temp2.addParameter<std::vector<std::string>>("RecHitFlags", {
        "",
      });
      temp2.addParameter<std::vector<std::string>>("ChannelStatus", {
        "HcalCellCaloTowerMask",
      });
      temp2.addParameter<int>("Level", 12);
      temp1.push_back(temp2);
    }
    {
      edm::ParameterSet temp2;
      temp2.addParameter<std::vector<std::string>>("RecHitFlags", {
        "",
      });
      temp2.addParameter<std::vector<std::string>>("ChannelStatus", {
        "HcalCellHot",
      });
      temp2.addParameter<int>("Level", 15);
      temp1.push_back(temp2);
    }
    {
      edm::ParameterSet temp2;
      temp2.addParameter<std::vector<std::string>>("RecHitFlags", {
        "",
      });
      temp2.addParameter<std::vector<std::string>>("ChannelStatus", {
        "HcalCellOff",
        "HcalCellDead",
      });
      temp2.addParameter<int>("Level", 20);
      temp1.push_back(temp2);
    }
    desc.addVPSet("SeverityLevels", vpsd1, temp1);
  }
  desc.add<std::vector<std::string>>("DropChannelStatusBits", {
    "HcalCellMask",
    "HcalCellOff",
    "HcalCellDead",
  });
  descriptions.add("hcalRecAlgos", desc);
}